File: 1304-find-n-unique-integers-sum-up-to-zero/1304-find-n-unique-integers-sum-up-to-zero.java
class Solution {
    public int[] sumZero(int n) {
        if(n == 1) return new int[]{0};
        int[] arr = new int[n];
        if(n%2 == 0){
            int j=1;
            for(int i=0;i<n;i+=2){
                arr[i] = j;
                arr[i+1] = -j;
                j++;
            }
        }else{
            int j=1;
            for(int i=0;i<n-1;i+=2){
                arr[i] = j;
                arr[i+1] = -j;
                j++;
            }
            arr[n-1] = 0;
        }
        return arr;
    }
}

File: 1304-find-n-unique-integers-sum-up-to-zero/1304-find-n-unique-integers-sum-up-to-zero.js
/**
 * @param {number} n
 * @return {number[]}
 */
var sumZero = function(n) {
    let arr = [];
    let i=1;
    if(n%2 == 0){
        for(let j=0;j<n;j+=2){
            arr[j] = i;
            arr[j+1] = -i;
            i++;
        }
    }else{
        for(let j=0;j<n-1;j+=2){
            arr[j] = i;
            arr[j+1] = -i;
            i++;
        }
        arr[n-1] = 0;
    }
    return arr;
};